server/routers/authenticate/authProviders/passkeys.ts

```diff
@@ -0,0 +1,714 @@
+// src/server/routers/passkeys.ts
+import {
+  generateRegistrationOptions,
+  verifyRegistrationResponse,
+  generateAuthenticationOptions,
+} from "@simplewebauthn/server";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { publicProcedure } from "@/server/trpc";
+import {
+  relyingPartyID,
+  relyingPartyName,
+  relyingPartyOrigin,
+} from "@/server/services/webauthnConfig";
+import { stringToUint8Array, uint8ArrayToString } from "@/server/services/auth";
+import { createServerClient } from "@/server/utils/supabase/supabase-server-client";
+import { createWebAuthnAccessTokenForUser, createWebAuthnRefreshTokenForUser } from "@/server/utils/passkey/session";
+import { getClientIp, getClientCountryCode } from "@/server/utils/ip/ip.utils";
+
+export const passkeysRoutes = {
+  // Start registration requiring a pre-existing user
```

Comment:
If it is required, why is there an `.optional()` below and why is there a cases in the function below to handle the case where the email is no provided? Isn't that the case we said was not possible with Supabase?